esx_service/volume_kv.py

```diff
@@ -42,24 +47,27 @@
 # The name of the VSAN policy applied to the VSAN volume. Invalid for Non-VSAN
 # volumes.
 VSAN_POLICY_NAME = 'vsan-policy-name'
+
+## Value for VSAN_POLICY_NAME
+DEFAULT_VSAN_POLICY = '[VSAN default]'
+
 # The size of the volume
 SIZE = 'size'
+DEFAULT_DISK_SIZE = "100mb"
 
 # The disk allocation format for vmdk
 DISK_ALLOCATION_FORMAT = 'diskformat'
-
 VALID_ALLOCATION_FORMATS = ["zeroedthick", "thin", "eagerzeroedthick"]
-
 DEFAULT_ALLOCATION_FORMAT = 'thin'
 
-## Values for given keys
-
-## Value for VSAN_POLICY_NAME
-DEFAULT_VSAN_POLICY = '[VSAN default]'
+# attach type. Default is independent.
+ATTACH_AS = 'attach_as'
 
-## Values for STATUS
-ATTACHED = 'attached'
-DETACHED = 'detached'
+# actual values to pass to VM reconfig events
+INDEPENDENT = 'independent_persistent'  # does not participate in vm snashot
+DEPENDENT   = 'persistent' # does participated in VM snapshot
```

Comment:
@Prashant - these are the actual strings which need to be passed to VMOMI: https://github.com/vmware/pyvmomi/blob/master/docs/vim/vm/device/VirtualDiskOption/DiskMode.rst. We can of course translate our string to VMOMI strings, but I feel it makes already muddy naming even worse

@Govindan - I agree it could (and should) be more user friendly and attach mode strings  is confusing.  But they are already used / documented in VMOMI specs. I will add a comment though,